### ISSUE TITLE
About The meta_info of Datasets

### ISSUE THREAD

**Issue by HankLiu10** [decisionintelligence/DUET#7]
Hi authors, thanks for your excellent work and the available code&data!
I found the datasets' length in your paper differs from your provided datasets. For example, the length of ETTh1/ETTh2 is 14400. While in your ETTh1.csv and other multivariate time series works, the length should be 17420. It seems that the number of instances in the train set, validation set , and test set is not properly set in the code. The portion of these 3 sets is actually 8640: 3392: 5900. I believe this problem also happens to some other datasets. Could you please check this potential issue? Thanks again for your great work!

**Comment by HankLiu10** [comment]
I believe this could probably be issued by the meta_info. For example, here's the meta_info of ETTh1.

![Image](https://github.com/user-attachments/assets/a1bd70bf-d296-41a9-962a-33ae2cd87675)

**Comment by qiu69** [comment]
Thank you for your attention and question. What you mentioned is indeed correct. Although the full sequence length of ETTh1.csv is 17,420, it is common practice in academia to use the first 14,400 data points, discarding the remainder. We also follow this convention, so the actual sequence length we use is 14,400. The data split ratio is 6:2:2, and this split is also based on the 14,400-length sequence.

**Comment by HankLiu10** [comment]
Thank you for your quick reply. I understand your explanation now. However, I printed out test_data.shape in the _eval_batch function as well as train_data_value.shape and valid_data.shape in the train_val_split function. I noticed that the lengths of the train, validation, and test sets are 8640, 3392, and 5900 respectively, which add up to 17,932 in total. Is there a potential issue here, or is this expected behavior?

**Comment by HankLiu10** [comment]
> Thank you for your attention and question. What you mentioned is indeed correct. Although the full sequence length of ETTh1.csv is 17,420, it is common practice in academia to use the first 14,400 data points, discarding the remainder. We also follow this convention, so the actual sequence length we use is 14,400. The data split ratio is 6:2:2, and this split is also based on the 14,400-length sequence.

Thank you for your quick reply. I understand your explanation now. However, I printed out test_data.shape in the _eval_batch function as well as train_data_value.shape and valid_data.shape in the train_val_split function. I noticed that the lengths of the train, validation, and test sets are 8640, 3392, and 5900 respectively, which add up to 17,932 in total. Is there a potential issue here, or is this expected behavior?

**Comment by qiu69** [comment]
Perhaps you made a mistake in your operation.
<img width="1183" alt="Image" src="https://github.com/user-attachments/assets/203d279e-8d21-4339-8e12-96c64de5cf81" />

**Comment by qiu69** [comment]
14400*0.6=8640, so train_lens=8640 is correct! validation_lens=3392, see screenshot. The same as test_len.

This means that the lookback window data in the first window of the validation set is obtained from the training set. Similarly, the lookback window data in the first window of the test set is obtained from the validation set.

<img width="754" alt="Image" src="https://github.com/user-attachments/assets/bd461075-abb5-4a2f-92b6-f613246799e1" />

<img width="1181" alt="Image" src="https://github.com/user-attachments/assets/1c90fde5-53f6-4713-b3f8-ab6416f0b6e3" />

**Comment by HankLiu10** [comment]
I understand you now. There's no problem here. 

Thanks!

**Comment by qiu69** [comment]
Thank you for your attention and question again！Best wishes！
